quantum/rgb_matrix.c

```diff
@@ -116,12 +112,12 @@ uint8_t rgb_frame_buffer[MATRIX_ROWS][MATRIX_COLS] = {{0}};
   static last_hit_t last_hit_buffer;
 #endif // RGB_MATRIX_KEYREACTIVE_ENABLED
 
-uint32_t eeconfig_read_rgb_matrix(void) {
-  return eeprom_read_dword(EECONFIG_RGB_MATRIX);
+void eeconfig_read_rgb_matrix(void) {
```

Comment:
These should really be moved to eeconfig.c tbh.

Reply:
Only problem with that, is that it is using the rgbmatrix structure, and that would have to be referenced in the eeconfic header file.

Reply:
None of the others do this. Not a fan of the inconsistency :/

Reply:
The functions in eeconfig.c have the data passed in the generic form uint8_t, uint16_t, and uint32_t all of which are data copies just to avoid what, avoid needing to use the address in the calling code? In the rgb matrix case the eeconfig_read_rgb_matrix function would still have to take a void pointer and size, which at that point, might as well take the address and leave the implementation / calling site in rgb matrix itself as they are internal / private.

Reply:
It would be good to have the rest of the eeconfig functions use their respective structs in that case, for type safety.

Reply:
while true, that is orthogonal to this pr.